statsproxy.go

```diff
@@ -47,134 +33,50 @@ func monitor() {
 		defer pprof.StopCPUProfile()
 	}
 
-	ticker := time.NewTicker(time.Duration(config.Service.TickerPeriod) * time.Second)
 signalLoop:
 	for {
 		select {
 		case sig := <-signalchan:
 			common.Logger.Info(fmt.Sprintf("!! Caught signal %d... shutting down\n", sig))
 			break signalLoop
-		case <-ticker.C:
-			sendStats()
+		case <-common.StatsTicker.C:
```

Comment:
Nope, doesn't block. At least, in my experience, everything works as intended, including the signal handler.